### DISCUSSION TITLE
array/object translations aren't being interpolated. is this inteded?

### DISCUSSION THREAD

**Discussion by dreyks** [fnando/i18n#18] [Q&A]
current version of `i18n-js` does the interpolations in any case

**Comment by fnando** [comment]
@dreyks can you give us more details? What does your translation file look like and how are you calling i18n?

**Reply by dreyks** [reply]
I'm setting the translations dynamically like this
```js
const i18n = new I18n()

i18n.store({
  en: {
    someKey: ['a string with an %{interpolation}', 'another string']
  }
})

console.log(i18n.t('someKey', { interpolation: 'interpolation value' })) // => ['a string with an %{interpolation}', 'another string'] i.e. not interpolated
```

this happens because only string results are being interpolated https://github.com/fnando/i18n/blob/8a403e00d3c9babc7f34dd2d647da9dd4da83b96/src/I18n.ts#L368-L369

while current i18n-js handles arrays and objects too https://github.com/fnando/i18n-js/blob/814c24981070eaf010c4bed4c5713282a019e838/app/assets/javascripts/i18n.js#L603-L611

**Reply by fnando** [reply]
I think the reason is that you're expecting arrays to be interpolated, which is not a feature supported by i18n-js. Instead, try something like:

```
> i18n.store({en: {someKey: "hello, %{name}", otherKey: "another one"}});
undefined
> i18n.t("someKey", {name: "John"})
'hello, John'
> i18n.t("otherKey")
'another one'
>
```

If you need an array out the translations, I recommend doing something like `keys.map(key => i18n.t(key, options))`.

**Reply by dreyks** [reply]
yeah i just wanted to figure out if it's not supported _yet_ or is it out of scope for this library as compared to the old i18n-js

**Reply by fnando** [reply]
This is tricky, because it breaks things that expect raw arrays. 🤔

**Reply by fnando** [reply]
As a quick thought, maybe we can introduce an option like `interpolateLists`, which would do that for you.

**Reply by dreyks** [reply]
by raw you mean uninterpolated?
